Src/ReadAccelGyroMagnetism.c

```diff
@@ -8,13 +8,164 @@
 
 static int READ_ACCEL_GYRO_MAGNETISM = 1000;
 
+static const int CMD_SIZE = 1;
+static const int CMD_TIMEOUT = 150;
+static const uint8_t RESET_CMD = 0x1E;
+static const uint8_t READX_LOW_CMD = 0x28;
```

Comment:
I don't think command is complete. 

From the datasheet https://drive.google.com/drive/folders/1Rj0pKpySH2fQZQkjZjC-vX5yY4cU9llD 
```
bit 0: READ bit. The value is 1.
bit 1-7: address AD(6:0). This is the address field of the indexed register.
```

The read bit is missing. So the math would be
```
readCmd = 1;
xLowRegisterAddress = 0x28;
cmd = xLowRegisterAddress << 1 | readCmd;
```